### PR TITLE
Sub project nstd::map unit test

### PR DESCRIPTION
**Pull request by yosmo78**
Closes N/A
Unit test for map

### Key features
* Unit test for nstd::map
* tests map storage, movement through map, and string hashing
* map unit test will be in a separate sub project

### Future Improvements
* more test possible test for the map

### Notes
none

### CODE REVIEW

File: tests/map_test/.gitignore [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# This file is used to ignore files which are generated
```
Comment (ChrisRzech) [inline tests/map_test/.gitignore]: Why is this `.gitignore` added?

File: tests/map_test/map_test.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
+#include <QtTest>
+#include <string>
+#include "csvparser.hpp"
+#include "src/utils/map.hpp"
+// add necessary includes here
```
Comment (ChrisRzech) [inline tests/map_test/map_test.cpp]: Remove

**Review by ChrisRzech — DISMISSED** [review]

File: tests/map_test/.gitignore (hunk 1 above)
Comment (yosmo78) [reply tests/map_test/.gitignore]: removed gitignore

File: tests/map_test/map_test.cpp (hunk 2 above)
Comment (yosmo78) [reply tests/map_test/map_test.cpp]: fixed commenting

**Review by ChrisRzech — APPROVED** [review]
👍

**Review by Roger-Chavez — APPROVED** [review]

**Pull request merged**